models/engine/db_storage.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+"""Definition of the DBStorage class."""
+from ast import literal_eval
+from models.base_model import Base
+from models.city import City
+from models.place import Place
+from models.review import Review
+from models.state import State
+from models.user import User
+from models.amenity import Amenity
+from models.base_model import BaseModel
+from os import getenv
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+
+
+class DBStorage:
+    """Represents a DBStorage object.
+
+    Args:
+        __engine: The current sqlalchemy engine.
+        __session: The current sqlalchemy session.
+    """
+
+    __engine = None
+    __session = None
+
+    def __init__(self):
+        """Initialiazes a database."""
+        self.__engine = create_engine("mysql+mysqldb://{}:{}@{}/{}"
+                                      .format(getenv("HBNB_MYSQL_USER"),
+                                              getenv("HBNB_MYSQL_PWD"),
+                                              getenv("HBNB_MYSQL_HOST"),
+                                              getenv("HBNB_MYSQL_DB")),
+                                      pool_pre_ping=True)
+
+        if getenv("HBNB_ENV") == "test":
+            Base.metadata.drop_all(self.__engine)
+
+    def all(self, cls=None):
+        """Returns a dictionary containing depending key: object
+           pairs on the cls provided."""
+        if cls:
```

Comment:
this I believe you meant to write `if not cls:` or `if cls is None:`

Reply:
Thanks for pointing that out :+1: